driver/src/contracts/mod.rs

```diff
@@ -3,16 +3,19 @@ pub mod stablex_auction_element;
 pub mod stablex_contract;
 
 use crate::error::DriverError;
+use crate::transport::LoggingTransport;
 use ethcontract::contract::MethodDefaults;
 use ethcontract::{ethsign, Account, SecretKey, H256};
+use log::Level;
 use std::env;
 use web3::api::Web3;
 use web3::transports::{EventLoopHandle, Http};
 
-fn web3_provider() -> Result<(Web3<Http>, EventLoopHandle), DriverError> {
+fn web3_provider() -> Result<(Web3<LoggingTransport<Http>>, EventLoopHandle), DriverError> {
```

Comment:
Do we need to be explicit about the type here? Couldn't it be an abstract Transport?

Reply:
We could make it `impl Transport`.

Reply:
Not making this a concrete type makes things more complicated so I'll leave it in for now.